src-java/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/FlowMirrorPathRepository.java

```diff
@@ -0,0 +1,38 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories;
+
+import org.openkilda.model.FlowMirrorPath;
+import org.openkilda.model.FlowPathStatus;
+import org.openkilda.model.PathId;
+import org.openkilda.model.SwitchId;
+
+import java.util.Collection;
+import java.util.Optional;
+
+public interface FlowMirrorPathRepository extends Repository<FlowMirrorPath> {
+    boolean exists(PathId mirrorPathId);
+
+    Collection<FlowMirrorPath> findAll();
+
+    Optional<FlowMirrorPath> findById(PathId pathId);
+
+    Optional<FlowMirrorPath> findByDestEndpoint(SwitchId switchId, int port, int vlan);
```

Comment:
Perhaps "egress" is more handy that "dest"? (just thoughts).

Reply:
Renamed.